www/artworks/post.php

```diff
@@ -0,0 +1,139 @@
+<? /** @noinspection PhpComposerExtensionStubsInspection, PhpIncludeInspection */
+
+use Ramsey\Uuid\Uuid;
+
+require_once('Core.php');
+
+/**
+ * @return array<string> an array containing [0] the path to the uploaded image and [1] the path to the thumbnail
+ * @throws \Exceptions\InvalidImageUploadException
+ */
+function handleImageUpload($uploadTmp): array {
+	$uploadInfo = getimagesize($uploadTmp);
+
+	if ($uploadInfo === false) {
+		throw new Exceptions\InvalidImageUploadException();
+	}
+
+	if ($uploadInfo[2] !== IMAGETYPE_GIF && $uploadInfo[2] !== IMAGETYPE_JPEG && $uploadInfo[2] !== IMAGETYPE_PNG) {
+		throw new Exceptions\InvalidImageUploadException();
+	}
+
+	$uid = Uuid::uuid4()->toString();
+	$ext = image_type_to_extension($uploadInfo[2]);
+
+	$uploadPath = COVER_ART_UPLOAD_PATH . $uid . $ext;
+	$thumbPath = COVER_ART_UPLOAD_PATH . $uid . '.thumb' . $ext;
+
+	if (!move_uploaded_file($uploadTmp, $uploadPath)) {
+		throw new Exceptions\InvalidImageUploadException();
+	}
+
+	$src_w = $uploadInfo[0];
+	$src_h = $uploadInfo[1];
+
+	if ($src_h > $src_w) {
+		$dst_h = COVER_THUMBNAIL_SIZE;
+		$dst_w = intval($dst_h * ($src_w / $src_h));
+	} else {
+		$dst_w = COVER_THUMBNAIL_SIZE;
+		$dst_h = intval($dst_w * ($src_h / $src_w));
+	}
+
+	switch ($uploadInfo[2]) {
```

Comment:
I think we should accept only JPEG uploads. It would simplify this code. Plus, the eventual ebook producer will need to supply a `cover.source.jpg`, so a PNG or GIF will be less useful.

Reply:
Good point, I've done that.